mlir/utils/jenkins/Jenkinsfile.miopen-test

```diff
@@ -1,65 +1,101 @@
 pipeline {
-    agent {
-        docker {
-            image 'rocm/mlir:miopen_test'
-            args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
-            label 'gfx908'
-        }
-    }
+    agent none
     stages {
-        stage('Environment') {
-            steps {
-                sh 'cat /etc/os-release'
-                sh '/opt/rocm/bin/rocm-smi'
-            }
-        }
-        stage('Build libMLIRMIOpen') {
-            steps {
-                checkout scm
-                sh '''
-                    mkdir -p build && cd build
-                    /usr/local/bin/cmake -G "Unix Makefiles" ../llvm \
-                         -DLLVM_ENABLE_PROJECTS="mlir;lld" \
-                         -DCMAKE_BUILD_TYPE=Release \
-                         -DBUILD_SHARED_LIBS=OFF \
-                         -DLLVM_BUILD_LLVM_DYLIB=OFF \
-                         -DLLVM_ENABLE_TERMINFO=OFF
-                    make -j$(nproc) libMLIRMIOpen
-                '''
+        stage("Carry out test") {
+            // Note: the builds are reused between cases on each node
+            // It may be possible to move build steps out to execute once
+            // but this could cause issues with target-specific code
+            // when we run the non-xdlops tests on a gfx900
+            matrix {
+                agent {
+                    docker {
+                        image 'rocm/mlir:miopen_test'
+                        args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
+                        label "${XDLOPS == '1' ? 'gfx908 && multi_gpu' : 'gfx908 && multi_gpu'}"
+                        // The above could have gfx900 || gfx908 in the latter half
```

Comment:
Have you tried to use gfx900 nodes in the latter half? I'm not sure we have such a lable, and the one I remember is just labeled as `rocm`.

Reply:
There is at least one node labelled gfx900, but I haven't tried this yet, since the dev machine didn't compile miopen (and it's gfx900)

Reply:
> since the dev machine didn't compile miopen

What's the problem with compiling miopen in gfx900?

Reply:
It's been fixed - it was a boost issue